src/host/emscripten/mod.rs

```
@@ -223,125 +216,152 @@ impl DeviceTrait for Device {
         // See also: The call to `set_timeout` at the end of the `audio_callback_fn` which creates
         // the loop.
         set_timeout(
-            || {
-                audio_callback_fn::<D, E>(
-                    user_data_ptr as *mut c_void,
-                    config,
-                    sample_format,
-                    buffer_size_frames,
-                )
-            },
             10,
+            stream.clone(),
+            data_callback,
+            config,
+            sample_format,
+            buffer_size_frames as u32,
         );
 
         Ok(stream)
     }
 }
 
+// TODO: Do something useful with the values, maybe through wasm_bindgen_futures
```

What do you mean by values here?

---

Sorry, I forgot to deal with that before committing. I was unsure about how to use a JS Promise and actually make sure it runs; I'll commit in a second a way using wasm-bindgen-futures to turn the JS Promise into a Rust future that we can then await and spawn. The only downside is that if the promise is rejected, we have to panic instead of actually returning an `Err`, unless we want to write a custom executor ourselves for this one usecase.